### PR TITLE
Import from https://github.com/darwinia-network/darwinia-2.0

### PR DESCRIPTION
**Pull request by AurevoirXavier**
- https://github.com/darwinia-network/darwinia-2.0/issues/9

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
@boundless-forest You changed the link. It became un-clickable.

**Comment by boundless-forest** [comment]
> @boundless-forest You changed the link to code. It became un-clickable.

It's such a huge change that I think you need to add some description there to explain what it does and why.

**Comment by AurevoirXavier** [comment]
> It's such a huge change that I think you need to add some description there to explain what it does and why.

Added https://github.com/darwinia-network/darwinia-2.0/issues/9 to the description.

**Comment by hackfisher** [comment]
Why there are these differences?

```
diff -bur darwinia/Makefile.toml darwinia-2.0/Makefile.toml
--- darwinia/Makefile.toml	2023-03-08 12:49:03.000000000 +0800
+++ darwinia-2.0/Makefile.toml	2023-03-08 12:39:13.000000000 +0800
@@ -113,12 +113,6 @@
 env     = { "FEATURES" = "all-natives,try-runtime" }
 extend  = "common-build"

-# Build for all runtimes.
-[tasks.build-ar]
-command = "cargo"
-env     = { "FEATURES" = "all-natives" }
-extend  = "common-build"
-
 # Build for testnets.
 [tasks.build-tns]
 command = "cargo"
diff -bur darwinia/tool/state-processor/src/staking/mod.rs darwinia-2.0/tool/state-processor/src/staking/mod.rs
--- darwinia/tool/state-processor/src/staking/mod.rs	2023-03-08 12:49:03.000000000 +0800
+++ darwinia-2.0/tool/state-processor/src/staking/mod.rs	2023-03-08 12:39:13.000000000 +0800
@@ -107,6 +107,17 @@
 				kton_pool += v.active_kton;
 				total_deposit += deposit_ring;
 				total_staked += v.active - deposit_ring;
+				total_staked += v
+					.ring_staking_lock
+					.unbondings
+					.iter()
+					.filter_map(
+						// Clear the expired unbondings.
+						//
+						// Since we don't add any lock here.
+						|u| if u.until == 0 { None } else { Some(u.amount) },
+					)
+					.sum::<Balance>();

 				self.shell_state.inc_consumers_by(&array_bytes::bytes2hex("", v.stash), consumers);
 				self.shell_state.insert_raw_key_value(
```

**Comment by AurevoirXavier** [comment]
> Why there are these differences?
> 
> Using
> 
> ```
> diff -bur darwinia/ darwinia-2.0/
> ```
> 
>  ```
> diff -bur darwinia/Makefile.toml darwinia-2.0/Makefile.toml
> --- darwinia/Makefile.toml	2023-03-08 12:49:03.000000000 +0800
> +++ darwinia-2.0/Makefile.toml	2023-03-08 12:39:13.000000000 +0800
> @@ -113,12 +113,6 @@
>  env     = { "FEATURES" = "all-natives,try-runtime" }
>  extend  = "common-build"
> 
> -# Build for all runtimes.
> -[tasks.build-ar]
> -command = "cargo"
> -env     = { "FEATURES" = "all-natives" }
> -extend  = "common-build"
> -
>  # Build for testnets.
>  [tasks.build-tns]
>  command = "cargo"
> diff -bur darwinia/tool/state-processor/src/staking/mod.rs darwinia-2.0/tool/state-processor/src/staking/mod.rs
> --- darwinia/tool/state-processor/src/staking/mod.rs	2023-03-08 12:49:03.000000000 +0800
> +++ darwinia-2.0/tool/state-processor/src/staking/mod.rs	2023-03-08 12:39:13.000000000 +0800
> @@ -107,6 +107,17 @@
>  				kton_pool += v.active_kton;
>  				total_deposit += deposit_ring;
>  				total_staked += v.active - deposit_ring;
> +				total_staked += v
> +					.ring_staking_lock
> +					.unbondings
> +					.iter()
> +					.filter_map(
> +						// Clear the expired unbondings.
> +						//
> +						// Since we don't add any lock here.
> +						|u| if u.until == 0 { None } else { Some(u.amount) },
> +					)
> +					.sum::<Balance>();
> 
>  				self.shell_state.inc_consumers_by(&array_bytes::bytes2hex("", v.stash), consumers);
>  				self.shell_state.insert_raw_key_value(
> ```

https://github.com/darwinia-network/darwinia-2.0/pull/321

**Comment by hackfisher** [comment]
Copy from this commit: https://github.com/darwinia-network/darwinia-2.0/commit/2f0b7cd4a426cbfb452706f1ae305243bf6edcf7

**Comment by AurevoirXavier** [comment]
First-time CI needs 1-2 hours.

**Comment by hackfisher** [comment]
New copy base: https://github.com/darwinia-network/darwinia-2.0/commit/63c91a8d474f9f30026a21a69c36c6777f20a80b

**Review by hackfisher — APPROVED** [review]

File: .github/release-template.md [hunk 1]
```diff
@@ -1,37 +1,42 @@
...
+| Pangolin | {{ pangolin_runtime_version }} |       LOW        |
+| Pangoro  | {{ pangoro_runtime_version }}  |       LOW        |
+
+## Resources
+
+### Pre-built Binary
+|  OS   |  Arch  | Glibc | LLVM  |                                                       Download                                                        |
+| :---: | :----: | :---: | :---: | :-------------------------------------------------------------------------------------------------------------------: |
+| Linux | x86_64 | 2.23  |  4.0  | [tar.zst](https://github.com/darwinia-network/darwinia-2.0/releases/download/{{ tag }}/darwinia-x86_64-linux-gnu.tar.zst) |
+| Linux | x86_64 | 2.23  |  4.0  | [tar.bz2](https://github.com/darwinia-network/darwinia-2.0/releases/download/{{ tag }}/darwinia-x86_64-linux-gnu.tar.bz2) |
```
Comment (boundless-forest) [inline .github/release-template.md:17]: Should we update this `https://github.com/darwinia-network/darwinia-2.0/releases/download`?

Comment (AurevoirXavier) [reply .github/release-template.md:17]: This PR is only about importing.

**Review by boundless-forest — APPROVED** [review]

**Pull request merged**